build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle

```diff
@@ -39,72 +40,7 @@ develocity {
-        uploadInBackground = false
-
-        String buildNumber = System.getenv('BUILD_NUMBER')
-        String buildUrl = System.getenv('BUILD_URL')
-        String jobName = System.getenv('JOB_NAME')
-        String nodeName = System.getenv('NODE_NAME')
-        String jobBranch = System.getenv('ghprbTargetBranch') ?: System.getenv('JOB_BRANCH')
-
-        // Link to Jenkins worker logs and system metrics
-        if (nodeName) {
-          link 'System logs', "https://ci-stats.elastic.co/app/infra#/logs?&logFilter=(expression:'host.name:${nodeName}',kind:kuery)"
-          buildFinished {
-            link 'System metrics', "https://ci-stats.elastic.co/app/metrics/detail/host/${nodeName}"
-          }
-        }
-
-        // Parse job name in the case of matrix builds
-        // Matrix job names come in the form of "base-job-name/matrix_param1=value1,matrix_param2=value2"
-        def splitJobName = jobName.split('/')
-        if (splitJobName.length > 1 && splitJobName.last() ==~ /^([a-zA-Z0-9_\-]+=[a-zA-Z0-9_\-&\.]+,?)+$/) {
-          def baseJobName = splitJobName.dropRight(1).join('/')
-          tag baseJobName
-          tag splitJobName.last()
-          value 'Job Name', baseJobName
-          def matrixParams = splitJobName.last().split(',')
-          matrixParams.collect { it.split('=') }.each { param ->
-            value "MATRIX_${param[0].toUpperCase()}", param[1]
-          }
-        } else {
-          tag jobName
-          value 'Job Name', jobName
-        }
-
-        tag 'CI'
-        link 'CI Build', buildUrl
-        link 'GCP Upload',
-          "https://console.cloud.google.com/storage/browser/_details/elasticsearch-ci-artifacts/jobs/${URLEncoder.encode(jobName, "UTF-8")}/build/${buildNumber}.tar.bz2"
-        value 'Job Number', buildNumber
-        if (jobBranch) {
-          tag jobBranch
-          value 'Git Branch', jobBranch
-        }
-
-        System.getenv().getOrDefault('NODE_LABELS', '').split(' ').each {
-          value 'Jenkins Worker Label', it
-        }
-
-        // Add SCM information
-        def isPrBuild = System.getenv('ROOT_BUILD_CAUSE_GHPRBCAUSE') != null
-        if (isPrBuild) {
-          value 'Git Commit ID', System.getenv('ghprbActualCommit')
-          tag "pr/${System.getenv('ghprbPullId')}"
-          tag 'pull-request'
-          link 'Source', "https://github.com/elastic/elasticsearch/tree/${System.getenv('ghprbActualCommit')}"
-          link 'Pull Request', System.getenv('ghprbPullLink')
-        } else {
-          value 'Git Commit ID', BuildParams.gitRevision
-          link 'Source', "https://github.com/elastic/elasticsearch/tree/${BuildParams.gitRevision}"
-        }
-      } else if (buildKiteUrl) { //Buildkite-specific build scan metadata
-        // Disable async upload in CI to ensure scan upload completes before CI agent is terminated
-        uploadInBackground = false
-
+      if (buildKiteUrl) { //Buildkite-specific build scan metadata
```

Comment:
Do we even need this conditional then? Given that if we are in CI, we are by definition running in Buildkite.

Reply:
I changed the conditional, but we still need to differ between ci and not ci. though you're right. CI basically means buildkite.